dao/mysql/transactions.go

```diff
@@ -0,0 +1,48 @@
+package mysql
+
+import (
+	"bonex-middleware/dao"
+	"bonex-middleware/dao/models"
+	"bonex-middleware/dao/mysql/driver"
+	sq "github.com/wedancedalot/squirrel"
+	"strings"
+)
+
+func (this *mysqlDAO) CreateTransaction(tx *models.Transaction, dao dao.DAOTx) error {
+	var err error
+
+	tx.Pubkey = strings.ToUpper(tx.Pubkey)
+	tx.Id, err = this.mysql.Insert(sq.Insert(models.TransactionsTable).SetMap(sq.Eq{
+		"mer_pubkey":     tx.Pubkey,
+		"mer_asset_code": tx.AssetCode,
+		"amount":         tx.Amount,
+		"secret":         tx.Secret,
+	}), this.daoTx2Sqlx(dao))
+
+	return err
+}
+
+func (this *mysqlDAO) RedeemTransaction(issuer, assetCode, secret string, dao dao.DAOTx) (*models.Transaction, error) {
```

Comment:
I'd highly recommend that any updates/deletes should always be performed by primary key.